test/solver/gmres_kernels.cpp

```diff
@@ -43,46 +43,46 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <ginkgo/core/base/executor.hpp>
 #include <ginkgo/core/base/math.hpp>
 #include <ginkgo/core/matrix/dense.hpp>
+#include <ginkgo/core/solver/gmres.hpp>
 #include <ginkgo/core/stop/combined.hpp>
 #include <ginkgo/core/stop/iteration.hpp>
 #include <ginkgo/core/stop/residual_norm.hpp>
 
 
-#include "core/solver/gmres_kernels.hpp"
 #include "core/test/utils.hpp"
+#include "test/utils/executor.hpp"
 
 
-namespace {
-
-
-class Gmres : public ::testing::Test {
+class Gmres : public CommonTestFixture {
 protected:
-#if GINKGO_DPCPP_SINGLE_MODE
-    using value_type = float;
-#else
-    using value_type = double;
-#endif
-    using index_type = gko::int32;
     using Mtx = gko::matrix::Dense<value_type>;
+    using Solver = gko::solver::Gmres<value_type>;
     using norm_type = gko::remove_complex<value_type>;
     using NormVector = gko::matrix::Dense<norm_type>;
     template <typename T>
     using Dense = typename gko::matrix::Dense<T>;
 
-    Gmres() : rand_engine(30) {}
-
-    void SetUp()
+    Gmres() : rand_engine(30)
     {
-        ASSERT_GT(gko::DpcppExecutor::get_num_devices("all"), 0);
-        ref = gko::ReferenceExecutor::create();
-        dpcpp = gko::DpcppExecutor::create(0, ref);
-    }
-
-    void TearDown()
-    {
-        if (dpcpp != nullptr) {
-            ASSERT_NO_THROW(dpcpp->synchronize());
-        }
+        mtx = gen_mtx(123, 123);
+        d_mtx = gko::clone(exec, mtx);
+        exec_gmres_factory =
+            Solver::build()
+                .with_criteria(
+                    gko::stop::Iteration::build().with_max_iters(246u).on(exec),
+                    gko::stop::ResidualNorm<value_type>::build()
+                        .with_reduction_factor(value_type{1e-15})
```

Comment:
still surprised at float germs can use 1e-15 as reduction_factor?

Reply:
yes, unit roundoff here. it's hard to reach that when in float.
I see, I miss the iteration criterion.